ibis/expr/types/generic.py

```diff
@@ -717,19 +719,16 @@ def substitute(
         └────────┴──────────────┘
         """
         if isinstance(value, dict):
-            expr = ibis.case()
-            try:
-                null_replacement = value.pop(None)
-            except KeyError:
-                pass
-            else:
-                expr = expr.when(self.isnull(), null_replacement)
-            for k, v in value.items():
-                expr = expr.when(self == k, v)
+            branches = sorted(value.items())
```

Comment:
Why are you sorting here?

Reply:
that is a great question lol. I'll remove it and see if any tests fail

Reply:
removed it.